scripts/lcov.sh

```diff
@@ -30,16 +30,7 @@ EOF
 
 set -eu
 
-# Project detection
-PROJECT_NAME_FILE='./scripts/project_name.txt'
-if read -r PROJECT_NAME < "$PROJECT_NAME_FILE"; then :; else
-    echo "$PROJECT_NAME_FILE does not exist... Exiting..." >&2
-    exit 1
-fi
-
-in_mbedtls_repo () {
-    test "$PROJECT_NAME" = "Mbed TLS"
-}
+. ../framework/scripts/project_detection.sh
```

Comment:
Sounds like a behaviour change to me: previously the script was expecting to be called from the project's root (ie `scripts/lcov.sh`) and now it seems to be expecting `cd scripts && ./lcov.sh`. Our Makefile [invokes it from the project's root](https://github.com/Mbed-TLS/mbedtls/blob/development/Makefile#L200) so we need to keep it working this way or this will break the nightly/release CI.

Generally speaking, scripts are expected to be invoked from the project's root (ie `scripts/lcov.sh`, `tests/scripts/all.sh` as opposed to `cd tests && scripts/all.sh` or `cd tests/scripts && ./all.sh`, etc.) Since explicit is better than implicit (`python -m this | sed -n 4p`), we might want to add more comments like [this one](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/scripts/all.sh#L32).